db/keyval/etcd/plugin_impl_etcd.go

```diff
@@ -131,12 +140,32 @@ func (plugin *Plugin) NewWatcher(keyPrefix string) keyval.ProtoWatcher {
 	return plugin.protoWrapper.NewWatcher(keyPrefix)
 }
 
+// DoResync performs ETCD resync
+func (plugin *Plugin) DoResync() {
+	plugin.Resync.DoResync()
+}
+
 // Disabled returns *true* if the plugin is not in use due to missing
 // etcd configuration.
 func (plugin *Plugin) Disabled() (disabled bool) {
 	return plugin.disabled
 }
 
+// Connected returns *true* if the plugin has connection with the database.
+func (plugin *Plugin) Connected() bool {
+	return plugin.connected
+}
+
+// GetInitNotificationChan returns post-init notification channel
+func (plugin *Plugin) GetInitNotificationChan() <-chan func() {
+	return plugin.initNotifChan
```

Comment:
This will not work very well with multiple plugins that use etcd as dependency. Only one of them will be informed and the rest will block forever.